### PR TITLE
HIGHLY optimises camera setup

### PR DESCRIPTION
**Pull request by Laboredih123**
[INTERNAL][PERFORMANCE]
## About the PR 
Highly optimises camera setup by removing strings and inlining the function. Also sprinkles some light optimisations elsewhere.

## Why's this needed? 
Speed is good.

### CODE REVIEW

File: code/modules/camera_coverage/datum/controller/camera_coverage.dm [hunk 1]
```diff
@@ -28,17 +28,25 @@ var/global/datum/controller/camera_coverage/camera_coverage_controller
 	var/lastpct = 0
 	var/thispct = 0
 	var/donecount = 0
+	var/totalcount = length(cam_candidates) / 100
```
Comment (ZeWaka) [inline code/modules/camera_coverage/datum/controller/camera_coverage.dm:31]:
```suggestion
	var/const/totalcount = length(cam_candidates) / 100
```

File: code/modules/camera_coverage/datum/controller/camera_coverage.dm [hunk 2]
```diff
@@ -61,11 +69,15 @@ var/global/datum/controller/camera_coverage/camera_coverage_controller
 /datum/controller/camera_coverage/proc/update_turfs(list/turf/turfs_to_update)
 	if (!length(turfs_to_update))
 		return
+
+	LAZYLISTINIT(src.turf_update_queue)
 	for(var/turf/T as anything in turfs_to_update)
 		if (global.explosions.exploding || ON_COOLDOWN(T, "camera_coverage_update", CAM_TURF_UPDATE_COOLDOWN))
-			LAZYLISTADDUNIQUE(src.turf_update_queue, turfs_to_update)
+			src.turf_update_queue |= turfs_to_update
```
Comment (ZeWaka) [inline code/modules/camera_coverage/datum/controller/camera_coverage.dm:76]: why change this from lazy add unique

File: code/modules/camera_coverage/datum/controller/camera_coverage.dm [hunk 3]
```diff
@@ -61,11 +69,15 @@ var/global/datum/controller/camera_coverage/camera_coverage_controller
...
 			return
 		T.aiImage?.loc = length(T.camera_coverage_emitters) ? null : T
+	if(!length(src.turf_update_queue))
+		src.turf_update_queue = null
```
Comment (ZeWaka) [inline code/modules/camera_coverage/datum/controller/camera_coverage.dm:80]: and why null

Comment (Laboredih123) [reply code/modules/camera_coverage/datum/controller/camera_coverage.dm:80]: no need for the list if nothing got into it

File: code/modules/camera_coverage/datum/controller/camera_coverage.dm (hunk 2 above)
Comment (Laboredih123) [reply code/modules/camera_coverage/datum/controller/camera_coverage.dm:76]: doing an if for a fuckton of turfs tends to be slow

File: code/modules/camera_coverage/datum/controller/camera_coverage.dm (hunk 3 above)
Comment (Laboredih123) [reply code/modules/camera_coverage/datum/controller/camera_coverage.dm:80]: or actually hm, should i just keep it

Comment (ZeWaka) [reply code/modules/camera_coverage/datum/controller/camera_coverage.dm:80]: well if it updates again it gotta remake it, why bother

**Pull request merged**